client/me/reauth-required/index.jsx

```diff
@@ -245,30 +275,33 @@ class ReauthRequired extends Component {
 		const hasSmsRecoveryNumber =
 			!! this.props?.twoStepAuthorization?.data?.two_step_sms_last_four?.length;
 
+		const currentAuthType = enhancedSecurity ? 'webauthn' : twoFactorAuthType;
+
 		return (
 			<Dialog
 				autoFocus={ false }
 				className="reauth-required__dialog"
 				isFullScreen={ false }
 				isVisible={ this.props?.twoStepAuthorization.isReauthRequired() }
 			>
-				{ isSecurityKeySupported && twoFactorAuthType === 'webauthn' ? (
+				{ isSecurityKeySupported && currentAuthType === 'webauthn' ? (
 					<SecurityKeyForm
 						loginUserWithSecurityKey={ this.loginUserWithSecurityKey }
 						onComplete={ this.refreshNonceOnFailure }
 					/>
 				) : (
-					this.renderVerificationForm()
+					this.renderVerificationForm( enhancedSecurity )
 				) }
 				<TwoFactorActions
-					twoFactorAuthType={ twoFactorAuthType }
+					twoFactorAuthType={ currentAuthType }
 					onChange={ this.handleAuthSwitch }
 					isSmsSupported={
 						method === 'sms' || ( method === 'authenticator' && hasSmsRecoveryNumber )
 					}
 					isAuthenticatorSupported={ method !== 'sms' }
 					isSmsAllowed={ shouldEnableSmsButton }
 					isSecurityKeySupported={ isSecurityKeySupported }
+					requireSecurityKeyOnly={ enhancedSecurity }
```

Comment:
Instead of this additional prop, can we change the values of the `is{Method}Supported` props to really say `false` if enhanced security is on? Then, for example, `isAuthenticationSupported` would mean simply "OTP is supported, period" and not the less clear "OTP is supported given that another prop is not `true`"